HomeScreenLib/src/main/java/com/example/homescreenlib/ui/HomeAdapter.kt

```diff
@@ -0,0 +1,48 @@
+package com.example.homescreenlib.ui
+
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.example.homescreenlib.data.Dashboard
+import com.example.homescreenlib.databinding.DashboardItemsLayoutBinding
+
+class HomeAdapter : ListAdapter<Dashboard, HomeAdapter.ViewHolder>(HomeAdapterDiffCallback()) {
+
+    class ViewHolder private constructor(private val binding: DashboardItemsLayoutBinding) :
+        RecyclerView.ViewHolder(binding.root) {
+        fun bind(item: Dashboard){
+            binding.dashboardItem = item
+            binding.executePendingBindings()
+        }
+
+        companion object{
```

Comment:
Why need companion object ?

Reply:
Created a static method to directly access to the outer class. 
we can directly write the layoutInflater inside the onCreateViewHolder like below 
override fun onCreateViewHolder(parent: ViewGroup, viewType: Int): ViewHolder {
        val layoutInflater = LayoutInflater.from(parent.context)
        val binding = DashboardItemsLayoutBinding.inflate(layoutInflater, parent, false)
        return ViewHolder(binding)
    }